packages/process/src/node/terminal-process.ts

```diff
@@ -46,6 +52,19 @@ export class TerminalProcess extends Process {
 
         this.logger.debug('Starting terminal process', JSON.stringify(options, undefined, 2));
 
+        // node-pty doesn't handle the `shell` option by default.
+        // work-around: spawn the shell ourselves.
+        if (options.options && options.options.shell) {
+            const command = options.command;
+            if (process.platform === 'win32') {
+                options.command = 'cmd';
+                options.args = ['/c', command];
+            } else {
+                options.command = 'sh';
+                options.args = ['-i', '-c', command];
```

Comment:
Might look awkward, [but Node does the same kind of check](https://github.com/nodejs/node/blob/master/lib/child_process.js#L474-L498).

Looking at it maybe we can borrow some things from it?

Reply:
Eventually, it would be nice to be able to specify which shell to use in the task configuration.  And maybe a preference that says which says to use by default for shell tasks, if unspecified.  Because sh often points to a very basic shell that might not have the features you are used to.

For example, brace expansion.  In bash:

```
$ echo abc{123,456}
abc123 abc456
```

And in dash (which sh points to by default on Debian:

```
$ echo abc{123,456}
abc{123,456}
```

And that's a feature people might want to use in task command lines.

But I think sh is a good default here, since it's the thing that is the most likely to be found everywhere.

The VSCode shell tasks appear to have a key for this:

```
            "options": {
                "shell": {
                    "executable": "...",
                    "args": []
                }
            }
```

Reply:
Should be able to accept this now :)

Will see how to add a test for it.

Reply:
If you really want to, I guess you could write your own "shell" (a small node.js script), specify it as the shell executable and somehow verify that it is indeed called with the right arguments.